core/modules/replica/ControllerApp.cc

```diff
@@ -50,6 +55,98 @@ string const description =
     " operations in a replication setup.";
 
 using namespace lsst::qserv::replica;
+namespace util = lsst::qserv::util;
+
+void printAsTable(SqlResultSet const& result,
+                  bool sqlShowType) {
+
+    cout << "  error:     " << result.error << "\n"
+         << "  hasResult: " << (result.hasResult ? "yes" : "no") << "\n"
+         << "  #fields:   " << to_string(result.fields.size()) << "\n"
+         << "  #rows:     " << to_string(result.rows.size()) << "\n";
+
+    if (result.hasResult) {
+
+        size_t const numRows = result.rows.size();
+        size_t const numColumns = result.fields.size();
+
+        vector<shared_ptr<vector<string>>> tableDataByColumns;
+        tableDataByColumns.reserve(numColumns);
+
+        for (size_t columnIdx = 0; columnIdx < numColumns; ++columnIdx) {
+            auto tableColumnPtr = make_shared<vector<string>>();
+            tableColumnPtr->reserve(numRows);
+            tableDataByColumns.push_back(tableColumnPtr);
+        }
+        for (auto const& row :  result.rows) {                
+            for (size_t columnIdx = 0; columnIdx < numColumns; ++columnIdx) {
+
+                auto const& field  = result.fields[columnIdx];
+                auto const& cell   = row   .cells [columnIdx];
+
+                auto const& tableColumnPtr = tableDataByColumns[columnIdx];
+                if (row.nulls[columnIdx]) {
+                    tableColumnPtr->push_back("NULL");
+                } else {
+                    if (sqlShowType) {
+                        string typeName ="UNKNOWN MYSQL TYPE:" + to_string(field.type);
+                        switch (field.type) {
```

Comment:
no default?

Reply:
This was fixed in another commit.

Reply:
`fixup`?

Reply:
The code was refactored and moved into a more appropriate class/method (where it must belong):
```
class SqlResultType {
    ...
    /// Structure Field stores a content captured from MYSQL_FIELD
    struct Field {
        ...
        int32_t     type;       /// Field type (see MySQL headers for enum enum_field_types)
        ...
        /// @return string representatio of the type
        std::string type2string() const;
    };
}
```